### ISSUE TITLE
rsync to MegaFuse directory

### ISSUE THREAD

**Issue by nono313** [matteoserva/MegaFuse#50]
Hi,
I am trying to use rsync to copy over a folder to the directory mounted by MegaFuse. 
At first, the copy seems to work but after some files, rsync stops with an error regarding the first file it tried to copy : 
`rsync: write failed on "/mnt/mega/.encfs6.xml": Operation not permitted (1)
rsync error: error in file IO (code 11) at receiver.c(393) [receiver=3.1.1]`

At the same time, I have a lot of errors like this in the console where I started MegaFuse : 
`searching node by path: /PATH
file FILENAME not found in MEGA`

Copying with a cp command seems to work fine so I am wondering why rsync is not working.

Thanks

**Comment by tYYGH** [comment]
Indeed, [in my experience](http://yalis.fr/cms/index.php/post/2016/01/08/Secure-personnal-backup-in-the-Cloud(s)-using-Linux), rsync does not work with Megafuse. However, rsync is not efficient over mounted filesystems because then it has to download every byte (unless it is asked to only compare mtime and size). Normal rsync operation is achieved with access to the remote server, either directly using the rsync protocol, or through an SSH connection, neither of which is provided by Mega.

**Comment by GeneThomas** [comment]
Hello,
      I am seeing the same problems with rsync. Your link does not seem to show that rsync does not work with Megafuse? rsync, even with local files (including mounted filesystems) will only compare the checksum (read every byte) if -c is specified, from man rsync: "-c, --checksum              skip based on checksum, not mod-time & size". I'm using **rsync -a /home/owner/tmp/Pictures/ /mega/Pictures** 
From the rsync man page introduction: "Rsync  finds  files that need to be transferred using a "quick check" algorithm (by default) that looks for files that have changed in size or in last-modified time."

I can replicate the problem:

On an empty Mega account, with just the intrduction .pdf file:

**MegaFuse -c /home/owner/.megafuse.conf 2>&1 | tee /tmp/MegaFuse.log**

conf and log files attached.

[megafuse.conf.zip](https://github.com/matteoserva/MegaFuse/files/1516310/megafuse.conf.zip)
[MegaFuse.log](https://github.com/matteoserva/MegaFuse/files/1516313/MegaFuse.log)

Then **rsync -a /home/owner/tmp/Pictures/ /mega/Pictures 2>&1 | tee /tmp/rsync.log** in another window, log attached.

[rsync.log](https://github.com/matteoserva/MegaFuse/files/1516314/rsync.log)

/home/owner/tmp/Pictures/ is zipped and attached.
[Pictures.zip](https://github.com/matteoserva/MegaFuse/files/1516316/Pictures.zip)

Please look into this problem as rsync and mega would be a fantastic backup solution.

Yours Sincerely,

Gene Thomas.
